server/src/main/java/org/elasticsearch/action/admin/indices/freeze/FreezeIndexRequestBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.freeze;
+
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequestBuilder;
+import org.elasticsearch.client.ElasticsearchClient;
+
+/**
+ * Builder for freeze index request
+ */
+public class FreezeIndexRequestBuilder extends
```

Comment:
crazy idea. Maybe we don't need all these actions here and rather go and collapse all the actions like:
 * OpenIndex
 *  CloseIndex
 * FreezeIndex
 * ThawIndex

into a single action, request, response and just use an Enum to signal what action we wanna do. This would prevent use from doing all this stuff over and over again. it might also simplify deprecation down the road?

Reply:
Ha, I had the same idea. We do something similar with shrink and split where they are "resize operations" carried by a single request object. I am fine with this being done in a subsequent change.